lib/ethers/transaction.ex

```diff
@@ -109,6 +123,43 @@ defmodule Ethers.Transaction do
     |> then(&(<<2>> <> &1))
   end
 
+  def decode(tx) when is_map(tx) do
+    tx_type =
+      case decode_key(tx, "type") do
+        2 -> :eip1559
+        1 -> :legacy
+        _ -> nil
+      end
+
+    tx_body =
+      %{
+        type: tx_type,
+        chain_id: decode_key(tx, "chainId"),
```

Comment:
Currently TCS expects `chain_id` to be hex, so if it's decoded here, we need to remember to remove decoding in TCS :) Maybe I miss it, are we missing `value` field? :P